plugins/kubernetes-backend/src/routes/resourcesRoutes.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  CompoundEntityRef,
+  parseEntityRef,
+  stringifyEntityRef,
+} from '@backstage/catalog-model';
+import { CatalogApi } from '@backstage/catalog-client';
+import { InputError, AuthenticationError } from '@backstage/errors';
+import express, { Request } from 'express';
+import { KubernetesObjectsProvider } from '../types/types';
+import { getBearerTokenFromAuthorizationHeader } from '@backstage/plugin-auth-node';
+
+export const addResourceRoutesToRouter = (
+  router: express.Router,
+  catalogClient: CatalogApi,
+  objectsProvider: KubernetesObjectsProvider,
+) => {
+  const getEntityByReq = async (req: Request<any>) => {
```

Comment:
Sidenote for @backstage/maintainers: this should totally be something we provide a utility for